apps/internal/base/internal/local/server.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Package local contains a local HTTP server used with interactive authentication.
+package local
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+)
+
+var okPage = []byte(`
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <title>Authentication Complete</title>
+</head>
+<body>
+    <p>Authentication complete. You can return to the application. Feel free to close this browser tab.</p>
+</body>
+</html>
+`)
+
+const failPage = `
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <title>Authentication Failed</title>
+</head>
+<body>
+	<p>Authentication failed. You can return to the application. Feel free to close this browser tab.</p>
+	<p>Error details: error %s error_description: %s</p>
+</body>
+</html>
+`
+
+// Result is the result from the redirect.
+type Result struct {
+	// Code is the code sent by the authority server.
+	Code string
+	// Err is set if there was an error.
+	Err error
+}
+
+// Server is an HTTP server.
+type Server struct {
+	// Addr is the address the server is listening on.
+	Addr     string
+	resultCh chan Result
+	s        *http.Server
+	reqState string
+}
+
+// New creates a local HTTP server and starts it.
+func New(reqState string) (*Server, error) {
+	var l net.Listener
+	var err error
+	for i := 0; i < 10; i++ {
```

Comment:
Any reason as to why we have 10 retries to listen? Just understanding where that number came from and some sort of documentation for maintenance

Reply:
This was just me being OVERLY cautious.  And I chose it out of my [expletive].  I figured if it tried 10 times to get a port and couldn't, that it was just broken.  

This would only happen if all 65535 ports were in use and you were trying to get one.  I have seen this happen at Google, but this is probably overkill.

Joel, if you want to kill it for 1 attempt, you can.

Reply:
Hahaha just was trying to understand where the magic number came from :) I am fine using 10, rest iterations will just be skipped if it finds it sooner !